### DISCUSSION TITLE
RFC: Back API data with persistence layer

### DISCUSSION THREAD

**Discussion by ryanolee** [ThioJoe/YT-Spammer-Purge#66] [Ideas]
# What
This issue raises the idea that it might be worth adding a persistence layer to the "comments fetch" stage of the of the comments scanning as it runs (based on something like the `https://docs.python.org/3/library/sqlite3.html`). So that results of API fetches can be persisted between script executions.

# Why
This would allow for a data set to be built up over multiple executions and reduce overall API usage. The idea being that upon scanning a video / part of the channel all comments fetched from the API would be stored for later use. This would mean that in theory as the purge script is more commonly used (when deleting spam comments). Spam comments could be deleted from a pool of every comment ever fetched by script (and not just the comments fetched by the "fetch" step)

As an extension of this it might be possible to fetch up to the last known comment from any given fetch on a per video basis. (Though this would not be ideal as I don't think it would track comment threads consistently)

# How
This could be done by possibly keeping a local representation of fetched comment state using a library like `sqlite3` (Which is part of the python standard lib).

**Comment by ryanolee** [comment]
Hi @ThioJoe ,
Please let me know if this sounds like it might be a useful feature. If it does would be happy to create a POC with some of the ideas discussed above 🚀

**Comment by ThioJoe** [comment]
I think this would make things unnecessarily complicated to be honest. Also moving to discussions.
